wasp-ai/src/client/pages/ResultPage.jsx

```diff
@@ -392,14 +392,19 @@ export function OnSuccessModal({ isOpen, setIsOpen, appGenerationResult }) {
     const logText = appGenerationResult?.project?.logs?.find((log) =>
       log.content.includes("tokens usage")
     )?.content;
-    const regex = /total tokens usage: ~(\d+(\.\d+)?)/i;
-    const match = logText?.match(regex);
-    if (match && match[1]) {
-      const num = Number(match[1]) * 1000;
-      if (Number.isInteger(num)) {
-        setNumTokensSpent(num);
+    
+    if (logText) {
+      const regex = /Total\s+tokens\s+usage\s*:\s*~\s*(\d+(?:\.\d+){0,1})\s*k\b/;
+      const match = logText.match(regex);
+
+      if (match) {
+        const num = parseFloat(match[1]);
+        setNumTokensSpent(num * 1000);
+      } else {
+        console.log('No match found');
```

Comment:
You could make this more descriptive: "Failed to parse total number of tokens used: no regex match."